parsl/monitoring/db_manager.py

```diff
@@ -389,22 +395,24 @@ def _migrate_logs_to_internal(self, logs_queue, queue_tag, kill_event):
     def _update(self, table, columns, messages):
         try:
             self.db.update(table=table, columns=columns, messages=messages)
-        except Exception:
+        except BaseException:
             self.logger.exception("Got exception when trying to update Table {}".format(table))
             try:
                 self.db.rollback()
             except Exception:
                 self.logger.exception("Rollback failed")
+            raise
 
     def _insert(self, table, messages):
         try:
             self.db.insert(table=table, messages=messages)
-        except Exception:
+        except BaseException:
             self.logger.exception("Got exception when trying to insert to Table {}".format(table))
             try:
                 self.db.rollback()
             except Exception:
                 self.logger.exception("Rollback failed")
+            raise
```

Comment:
_insert and _update now pass on database errors to their caller, rather than absorbing them.
So probably the caller (the main loop process) now needs to deal with non-KeyboardInterrupt exceptions that might occur. Or, the _insert and _update code should only re-raise KeyboardInterrupt exceptions to preserve previous behaviour.

Reply:
I am choosing the later approach: raise KeyboardInterrupt exceptions in `_insert` and `_update` code. I think for exceptions other than KeyboardInterrupt, we should re-raise too since the db already misses some info at that point.

Reply:
> we should re-raise too since the db is missing some messages at that point, no?

the behaviour in master is to ignore most exceptions at the top level, and then carry on the main loop receiving and processing messages. I think this PR should not change that behaviour.

It would be worth investigating separately though.

Reply:
that makes sense. thanks